src/main/java/ox/app/utility/SetupChooser.java

```diff
@@ -0,0 +1,37 @@
+package ox.app.utility;
+
+import ox.app.exceptions.WrongArgumentException;
+
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public class SetupChooser {
+    private static int iterator = 3;
```

Comment:
Not the best name.
You should say for what collection this iterator is provided

Reply:
Changed.